virtcontainers/container.go

```diff
@@ -984,7 +984,8 @@ func (c *Container) update(resources specs.LinuxResources) error {
 
 	newResources := ContainerResources{
 		VCPUs: uint32(utils.ConstraintsToVCPUs(*resources.CPU.Quota, *resources.CPU.Period)),
-		MemMB: uint32(*resources.Memory.Limit >> 20),
+		// do page align to memory, as cgroup memory.limit_in_bytes will be aligned to page when effect
+		Mem: (*resources.Memory.Limit >> 12) << 12,
```

Comment:
see https://github.com/kata-containers/runtime/pull/786#discussion_r220417518

Reply:
That doesn't answer my question. The size is aligned with `memoryBlockSize` right after the page size alignment here. What is the point of having two alignments? The `memoryBlockSize` is multiple times of page size. I don't see why you need to align with page size first.

Reply:
In the case we don't do page size alignment first, if we setup a container of 100MB, then we update the container memory of 100MB+1B. As a result, we will hotplug 1 `memoryBlockSize` memory to vm, which is not recommended. Because for container memory limit cgroup, it would be 100MB rather than 100MB+1B when you update 100MB+1B to a 100MB memory limit container(linux kernel will do page alignment when memory limit cgroup takes effect, [reference](https://elixir.bootlin.com/linux/latest/source/mm/memcontrol.c#L2942)). So we don't have to hotplug a `memoryBlockSize` memory to vm for 1B.

If we do page align here, we will ignore 1B, but not ignore 4096B. It means we can hotplug a `memoryBlockSize` memory to vm if you update 100MB+4096B to a 100MB memoy limit container.

Reply:
Thanks for the explanation. Now I see you point. You want to avoid wasting the extra 1 `memoryBlockSize` of memory in case of 100MB + 1B.  And I think you should call `os.GetPageSize()` to find out the system page size rather than always assuming 4KB page size. For example, aarch64 might use 64KB page size.

Reply:
Yes, another field in the `GetGuestDetails` call ;)

Reply:
> And I think we should call `GetGuestDetails` when making the very first connection to the agent, and save all the fields we are interested in. Then we do not need to call it again at different places when these fields are needed.

Yes, this is what we do now. We only  `GetGuestDetails` when create sandbox and save all the fields in sandbox state.json. I will add a TODO comment in the code later.